scripts/integration-tests/default.nix

```diff
@@ -0,0 +1,46 @@
+{
+  nixpkgs ? (import ../../.).nixpkgs,
+  binaries ? (import ../../.).binaries,
+  haddock ? (import ../../.).haddock,
+  content ? (import ../../.).content,
+  system ? builtins.currentSystem,
+  lib ? nixpkgs.lib
+}:
+
+let
+  runCuriosity = nixpkgs.writers.writeBashBin "run-curiosity" ''
+    set -euo pipefail
+    PATH=${lib.strings.makeBinPath [ binaries ]}:$PATH
+    STATIC_PATH=${content}
+    ${builtins.readFile ./run-curiosity.sh}
+  '';
+  nginxConf = nixpkgs.substituteAll {
+    src = ./nginx.conf;
+    curiosityaddr = "http://127.0.0.1:9100";
+    # TODO: find a way to reference this folder without the explicit
+    #       0.1.0.0 version
+    curiosityhaddock = "${haddock.doc}/share/doc/curiosity-0.1.0.0/html";
+  };
+  runNGinx = nixpkgs.writers.writeBashBin "run-nginx" ''
+    set -euo pipefail
+    rundir=$(mktemp -d)
+    trap 'rm -rf -- "$rundir"' EXIT
+    cd "$rundir"
+    cp ${nginxConf} nginx.conf
+    mkdir -p nginx-cache
+    sed -i "s~@tmpdir@~$rundir/nginx-cache~g" nginx.conf
```

Comment:
Would it be possible to make `nginxConf` a function that takes the cache directory as argument, so that all the substitutions can be done by `substituteAll`, instead of having this little `sed` call here ?

Reply:
I think it's necessary:

- `substituteAll` injects the paths we can resolve **before** running the script. Here the haddock nix-store path.
- The `sed` call injects the paths depending on the script runtime (the `mktemp` call here).

----------------

It'd be possible to inject the runtime path to the `nginxConf` function, however, this would need to happen in a second Nix build triggered by the bash script itself.

IE. we would move from this exec flow:

1. Inject nix-specific path to the `run-nginx` script through a nix-build.
2. Run the resulting script.

to

1. Inject nix-specific path to the `run-nginx` script through a nix-build.
2. Run the resulting script.
3. Call `nix-build` from this script to generate the nginx-conf.

I'd prefer keeping a simple workflow for now.